java/connector-node/tracing/src/main/java/com/risingwave/tracing/TracingSlf4jAdapter.java

```diff
@@ -28,6 +28,42 @@ public class TracingSlf4jAdapter implements Logger {
 
     private final String name;
 
+    void logIfEnabled(int level, String msg) {
+        if (TracingSlf4jImpl.isEnabled(level)) {
```

Comment:
It seems unnecessary to check the logging level every time we write a log (two JNI calls for each log). IIUC the level is fixed after CN started, could we read the logging level once in the constructor of `TracingSlf4jAdapter` and store it in memory?

Reply:
This might not be true considering #11514. But since it's not implemented yet, memoizing them also sounds reasonable.

May I ask what's the extra cost for crossing the JNI boundary? Note that every time an event is generated with macros like `tracing::info!` in Rust codes, it'll also check whether the level is enabled. So I guess the checking itself is performant.

Reply:
The cost of crossing the JNI boundary can not be ignored if it's frequently called. I think we can load and store the log level setting at initialization and always read the stored value. In the future if we want to support changing the log level on the flight, we can spawn a thread to periodically load the latest log level setting and atomically update the stored value.

Reply:
> May I ask what's the extra cost for crossing the JNI boundary?

It really depends on how frequent we execute JNI calls, our existing scenario usually in stream chunk granularity. But for the case of `TracingSlf4jAdapter` it will intercepts all logging in the Java connector node, for example the `trace` log in the issue will be called for each record the connector snapshotted from upstream db. So I think we should optimize it.